server/sonar-server/src/main/java/org/sonar/server/project/ws/CreateAction.java

```diff
@@ -0,0 +1,140 @@
+  private final UserSession userSession;
+  private final ComponentService componentService;
+  private final DefaultOrganizationProvider defaultOrganizationProvider;
+  private final PermissionTemplateService permissionTemplateService;
+  private final FavoriteUpdater favoriteUpdater;
+
+  public CreateAction(DbClient dbClient, UserSession userSession, ComponentService componentService, PermissionTemplateService permissionTemplateService,
+    FavoriteUpdater favoriteUpdater, DefaultOrganizationProvider defaultOrganizationProvider) {
+    this.dbClient = dbClient;
+    this.userSession = userSession;
+    this.componentService = componentService;
+    this.permissionTemplateService = permissionTemplateService;
+    this.favoriteUpdater = favoriteUpdater;
+    this.defaultOrganizationProvider = defaultOrganizationProvider;
+  }
+
+  @Override
+  public void define(WebService.NewController context) {
+    WebService.NewAction action = context.createAction(ACTION_CREATE)
+      .setDescription("Create a project.<br/>" +
+        "Requires 'Create Projects' permission")
+      .setSince("4.0")
+      .setPost(true)
+      .setResponseExample(getClass().getResource("create-example.json"))
+      .setHandler(this);
+
+    action.createParam(ProjectsWsParameters.PARAM_KEY)
+      .setDescription("Key of the project")
+      .setRequired(true)
+      .setExampleValue(KEY_PROJECT_EXAMPLE_001);
+
+    action.createParam(PARAM_NAME)
+      .setDescription("Name of the project")
+      .setRequired(true)
+      .setExampleValue("SonarQube");
+
+    action.createParam(PARAM_BRANCH)
+      .setDescription("SCM Branch of the project. The key of the project will become key:branch, for instance 'SonarQube:branch-5.0'")
+      .setExampleValue("branch-5.0");
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    userSession.checkPermission(PROVISIONING);
+    CreateRequest createRequest = toCreateRequest(request);
+    writeProtobuf(doHandle(createRequest), request, response);
+  }
+
+  private CreateWsResponse doHandle(CreateRequest request) {
+    String organizationUuid = defaultOrganizationProvider.get().getUuid();
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      ComponentDto componentDto = componentService.create(dbSession, newComponentBuilder()
+        .setOrganizationUuid(organizationUuid)
+        .setKey(request.getKey())
+        .setName(request.getName())
+        .setBranch(request.getBranch())
+        .setQualifier(PROJECT)
+        .build());
+      handlePermissionTemplate(dbSession, componentDto, organizationUuid);
+      return toCreateResponse(componentDto);
+    }
+  }
+
+  private void handlePermissionTemplate(DbSession dbSession, ComponentDto componentDto, String organizationUuid) {
+    permissionTemplateService.applyDefault(dbSession, organizationUuid, componentDto, userSession.isLoggedIn() ? userSession.getUserId().longValue() : null);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') NullPointerException might be thrown as 'getUserId' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)